src/pooling.rs

```diff
@@ -0,0 +1,38 @@
+use ndarray::{s, ArrayBase, Dim, IxDynImpl, OwnedRepr, ViewRepr};
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+pub struct PoolingConfig {
+    pub pooling_mode_cls_token: bool,
+    pub pooling_mode_mean_tokens: bool,
+    pub pooling_mode_max_tokens: bool,
+    pub pooling_mode_mean_sqrt_len_tokens: bool,
+}
+
+pub enum LoadPoolingError {
+    FailToLoadCacheConfigFile,
+    FailToDeserialiseConfig,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Pooling {
+    Cls,
+    Mean,
+}
```

Comment:
My intention with none None currently is that there is no default pooling, so we fall back to the Pooling:Cls. 

If we were to add Pooling::None, it means that we shouldn't pool the output at all, In such case, I will update its behaviour in text_embedding.rs line 337 to force the ort output into a 2 dimension vec for return or throw if the shape is mismatch. What do you think @Anush008 ?

Reply:
> If we were to add Pooling::None, it means that we shouldn't pool the output at all, In such case, I will update its behaviour in text_embedding.rs line 337 to force the ort output into a 2 dimension vec for return or throw if the shape is mismatch. What do you think @Anush008 ?

TBH,  I have no preference, since I don't have expertise in ML. I am happy to go with your preference.

Reply:
Let's leave it for now. I would only be confident forcing the tensor shape down to vector with good tests. I haven't come across a model that has built in pooling yet. So it hard to write a sensible tests. 

I can easily add it later if we find one. I'll leave the todo in there for now.